core/src/core/pmmr.rs

```diff
@@ -661,6 +512,30 @@ pub fn peaks(num: u64) -> Vec<u64> {
 	peaks
 }
 
+/// Helper function to convert an index to a leaf index (the nth leaf)
+pub fn leaf_index(n: u64) -> u64 {
+	let mut count = 1;
+	let mut cur_leaf = n;
+
+	// Special case that causes issues in bintree functions,
+	// just return
+	if n == 1 {
+		return count;
+	}
+
+	loop {
+		if bintree_postorder_height(cur_leaf) > 0 {
+			cur_leaf = bintree_rightmost(cur_leaf);
+		}
+		count = count + 1;
+		cur_leaf = bintree_jump_left_sibling(cur_leaf);
+		if cur_leaf == 1 {
+			break;
+		}
+	}
+	count
+}
```

Comment:
Isn't this O(N) where N is the number of leafs in the tree? Each operation is fast but on a big tree it won't be too good...

Reply:
Yes, sorry I'd meant to mention this in my description.. it is temporary O(n) crap. However, the peaks() function we'd talked about earlier was returning strange values (or I'd quite possibly just misunderstood) and the method we'd talked about earlier wasn't doing what I was expecting, so I just implemented this quickly and moved on to other parts. This obviously needs to be changed.